File: mobile/test/core/model/model_test.dart
import 'dart:typed_data';

import 'package:meshsetu_mobile/core/model/model.dart';
import 'package:test/test.dart';

void main() {
  test('rejects expired envelope', () {
    expect(
      () => MeshEnvelope(
        objectId: 1,
        eventId: 'e',
        siteId: 's',
        roomId: 'r',
        createdAtMs: 10,
        expiresAtMs: 10,
        hopCount: 0,
        hopLimit: 1,
        priority: PriorityBand.p0Critical,
        payloadType: PayloadType.ack,
        payload: Uint8List.fromList([1]),
        originEphemeralId: 2,
      ),
      throwsArgumentError,
    );
  });
}

File: core-model/src/test/kotlin/in/meshsetu/model/ModelTest.kt
package `in`.meshsetu.model

import kotlin.test.Test
import kotlin.test.assertFailsWith

class ModelTest {
    @Test fun rejectsExpiredEnvelope() = assertFailsWith<IllegalArgumentException> {
        MeshEnvelope(1u, "e", "s", "r", 10, 10, 0, 1, PriorityBand.P0_CRITICAL, PayloadType.ACK, byteArrayOf(1), 2u)
    }
}
